### PR TITLE
Fixes #61 And fixes the regex to correctly handle constant assignment…

### LINKED ISSUE #61: Command type detection bug
Test case - `Reference.new == Reference.new` 
Even though this is an equality check, this is being caught as a `constant_assignment` command and so is wrongly treated. You do not get the output of the command.

### PR DESCRIPTION
**Pull request by ktaragorn**
… detection in general

Also widen tests available to cover various cases

### CODE REVIEW

File: spec/integration/icr_spec.cr [hunk 1]
```diff
@@ -260,6 +262,40 @@ describe "icr command" do
...
+      A = 0
+      B=1
+      HTTP_STATUS    =    404
+      Constant = "cheese"
+      ISO8859_1 = :latin
+      A =~ /test/
+      Abc_B = "123"
+      Abc_B =~ /^\d+/
+      CRYSTAL
+      icr(input).should match /0/
```
Comment (veelenga) [inline spec/integration/icr_spec.cr]: I don't think `.should match /0/` is ok here. Remove all lines before `Abc_B = "123"` or adjust the matcher accordingly.

File: spec/integration/icr_spec.cr [hunk 2]
```diff
@@ -260,6 +262,40 @@ describe "icr command" do
...
+    it "catches constant equality with constants with lower case letters" do
+      input = <<-CRYSTAL
+      Aa = 0
+      Bb = 1
+      Aa == Bb
+      CRYSTAL
+      icr(input).should match /false/
+    end
+
+    it "doesnt catch object comparison by .new" do
```
Comment (veelenga) [inline spec/integration/icr_spec.cr]: "does not..."

File: spec/integration/icr_spec.cr [hunk 3]
```diff
@@ -260,6 +262,40 @@ describe "icr command" do
...
+      input = <<-CRYSTAL
+        Reference.new == Reference.new
+      CRYSTAL
+      icr(input).should match /false/
+      input = <<-CRYSTAL
+        Reference.new != Reference.new
+      CRYSTAL
+      icr(input).should match /true/
```
Comment (veelenga) [inline spec/integration/icr_spec.cr]: one like inputs can be written without heredoc:

```crystal
icr("Reference.new == Reference.new").should match /false/
icr("Reference.new != Reference.new").should match /false/
```

**Comment by ktaragorn** [comment]
Good Catches.. Fixed!

**Comment by veelenga** [comment]
👍  /cc @jwoertink @greyblake

**Comment by jwoertink** [comment]
I checked it out, and it looks good 👍

**Pull request merged**